Docs/Book/GettingStartedWithContributing.md

```diff
@@ -0,0 +1,633 @@
+    ```
+- create and move to the build directory:
+    ```
+    mkdir build
+    cd build
+    ```
+- run cmake with the relevant flags e.g.
+  ```
+  cmake -DRDK_BUILD_PYTHON_WRAPPERS=OFF ..
+  ```
+  if you don't want to build the Python wrappers. The full list of options can be found in the [CMakeLists.txt](https://github.com/rdkit/rdkit/blob/master/CMakeLists.txt) file.
+- run make:
+  ```
+  make install -j10
+  ```
+- run the tests to check the build has been successful:
+  ```
+  RDBASE=$RDBASE DYLD_FALLBACK_LIBRARY_PATH="$RDBASE/lib:$PYROOT/lib" PYTHONPATH=$RDBASE ctest -j 10
+  ```
+
+Note: on MacOS for v14.0 onwards some of the changes caused problems with the standard CMake spell. This [discussion thread](https://github.com/rdkit/rdkit/discussions/6859) provides some useful solutions in this case.
+
+### Where Should I Put My Code? 
+
+The RDKit C++ code lives in `rdkit/Code`, with the majority of the functionality sitting in the `GraphMol` sub-directory. If the code being contributed interfaces with another project it should go in `rdkit/External` and in an appropriate sub-directory. Again if you are unsure where your submission should go check with the maintainers!
+
+### Coding Standards
+
+As discussed in the Python section above, there is no official formal style guide for RDKit, but we recommend following the suggestions below:
+- Generally the toolkit uses camelCase. Class and namespace names start with an upper-case letter (e.g. ExplicitBitVect), function and method names starting with a lower-case letter (e.g. getNumAtoms()).
+- Local variable names are usually lower-case, but sometimes use either snake_case or camelCase.
+- We follow the formatting suggestions from [Google's C++ style guidelines](https://google.github.io/styleguide/cppguide.html).
+- Use of an auto-formatter (e.g. [clang-format](http://clang.llvm.org/docs/ClangFormat.html)) is strongly recommended. This can be integrated with editors, but note that at least version 3.8 of clang-format is required as the configuration files are not backwards compatible. There is a .clang-format config file at the root of the repository; code layout should be based on this configuration.
+- The public classes / functions exposed in the API are typically defined in the header file, and the underlying method in the C++ file of the same name.
+- Use multi-line comments in the header file to document code:
+    ```c++
+    /*!
+    this is a documentation string
+    */
+    ```
+
+We would also recommend checking your code for memory leaks with [Valgrind](https://valgrind.org/) before opening a pull request.
+
+### Tests
+
+The RDKit C++ Tests use the [Catch2 framework](https://github.com/catchorg/Catch2). The tests generally are placed in a file named "catch_tests.cpp" in the same directory as the corresponding code. Small changes and bug fixes may only need the addition of a new TEST\_CASE to an existing test set rather than a new file.
```

Comment:
```suggestion
The RDKit C++ Tests use the [Catch2 framework](https://github.com/catchorg/Catch2). The tests generally are placed in a file named "catch_tests.cpp" in the same directory as the corresponding code. Unless you are adding a significant new feature, it's generally best to add the new TEST\_CASE to an existing test set rather than creating an entirely new file.
```